examples/nytimes.js

```diff
@@ -1,25 +1,35 @@
-const { firefox } = require("playwright");
-
-(async () => {
-  const browser = await firefox.launch();
-  const context = await browser.newContext();
-
-  const page = await context.newPage();
-
-  await page.goto("https://www.nytimes.com/");
-
-  await page.click(
-    'div[data-testid="masthead-mini-nav"] li[data-testid="mini-nav-item"] >> text="World"'
-  );
-
-  await page.click('button[data-test-id="search-button"]');
-
-  await page.fill('input[data-testid="search-input"]', "climate");
-
-  await page.press('input[data-testid="search-input"]', "Enter");
-
-  await page.close();
-
-  await context.close();
-  await browser.close();
-})();
+const { example, action } = require("../src/helpers");
+
+const selectors = {
+  nav: 'div[data-testid="masthead-mini-nav"]',
+  navItem: '[data-testid="mini-nav-item"]',
+  navItemByText(text) {
+    return `${this.nav} ${this.navItem} >> text="${text}"`;
+  },
+  searchButton: 'button[data-test-id="search-button"]',
+  searchInput: "input[data-testid='search-input']",
+  menuButton: "#desktop-sections-button",
+  menuItemByText(text) {
+    return `[data-testid="desktop-nav"] a[data-name="${text}"]`;
+  },
+};
+
+example("nytimes.com", async (page, { action }) => {
+  // using a lighter-weight starting page than the root
+  await page.goto("https://www.nytimes.com/sitemap");
```

Comment:
smart

Reply:
I noticed today we can add routing rules to suppress some HTTP requests which will be helpful for wrangling some of the heavier sites.

https://playwright.dev/docs/1.8.0/api/class-page#pagerouteurl-handler